packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -740,3 +832,113 @@ String getGradlewFileName(Platform platform) {
     return 'gradlew';
   }
 }
+
+// List of compatible Java/Gradle versions, where javaMax versions are
+// exclusive.
+//
+// Should be updated when a new version of Java is supported by a new version
+// of Gradle, as https://docs.gradle.org/current/userguide/compatibility.html
+// detials.
+List<JavaGradleCompat> _javaGradleCompatList = const <JavaGradleCompat>[
```

Comment:
This makes sense to me and would improve readability IMO. I am hesitant to make this change in this PR just because it is balloon-ing a little, but I can definitely follow up on this PR making this change.